### PR TITLE
[CCM] add AWS metric and tag source documentation

### PR DESCRIPTION
**Pull request by enemyghost**
### What does this PR do? What is the motivation?
- adding new metrics
- improving metric descriptions + providing advice
- adding tag sources

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/cloud_cost_management/aws.md [hunk 1]
```diff
@@ -135,20 +135,104 @@ Attach the new S3 policy to the Datadog integration role.
...
 
-### Cost types
+## Cost types
 
-You can visualize your ingested data using the following cost types:
+Visualize your ingested data using out-of-the-box cost types. Cloud cost metric names contain various terms that describe how they are calculated.
 
-| Cost Type            | Description           |
+### Amortized vs. unblended costs
+**Amortized** cost metrics distribute some pre-payments (like up-front savings plans fees) across the entire discount term. This is also called _accrual basis_. As savings plans are applied to covered usage, the costs for that savings plan appear as usage charges. The unused portion appears as a fee on the accrual date. Not all one-time fees are amortized across the billing period. For example, the unused portion of an RI Fee always appears on the first of the month.
```
Comment (glebec) [inline content/en/cloud_cost_management/aws.md:143]: > accrual date

Is the reader expected to know what "accrual date" means? Is it the date that a payment was applied?

Comment (glebec) [reply content/en/cloud_cost_management/aws.md:143]: The sentence starting "Not all one-time fees..." feels like it could be a second short paragraph.

Comment (glebec) [reply content/en/cloud_cost_management/aws.md:143]: What is an "RI Fee"?

**Comment by glebec** [comment]
I think this helps to clarify a lot; one thing I noticed is that the "on-demand" cost isn't explained until the Example section. Maybe it could get a short explanation before then?

**Review by glebec — APPROVED** [review]

File: content/en/cloud_cost_management/aws.md [hunk 2]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+## Cost types
 
-You can visualize your ingested data using the following cost types:
+Visualize your ingested data using out-of-the-box cost types. Cloud cost metric names contain various terms that describe how they are calculated.
 
-| Cost Type            | Description           |
+### On-demand
+**On-demand** costs are the costs of usage at the public, on-demand rate published by AWS. This excludes all savings plans, reservations, discounts, taxes, and fees.
+
+These costs do not provide an accurate representation of what is paid to AWS, so it's not recommended to use this metric in most cases.
```
Comment (urseberry) [inline content/en/cloud_cost_management/aws.md:145]:
```suggestion
On-demand costs do not accurately represent what you pay AWS. Do not use this metric to estimate your actual costs.
```

File: content/en/cloud_cost_management/aws.md [hunk 3]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+
+### Amortized vs. unblended costs
```
Comment (urseberry) [inline content/en/cloud_cost_management/aws.md:147]:
```suggestion
### Amortized and unblended costs
```

File: content/en/cloud_cost_management/aws.md [hunk 4]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+**Amortized** cost metrics distribute some pre-payments (like up-front savings plan fees) across the entire discount term. This is also called _accrual basis_. As savings plans are applied to covered usage, the costs for that savings plan appear as usage charges. The unused portion appears as a fee.
```
Comment (urseberry) [inline content/en/cloud_cost_management/aws.md:148]:
```suggestion
**Amortized** cost metrics distribute some pre-payments (such as up-front savings plan fees) across the entire discount term. Another name for amortized cost metrics is _accrual basis_. As savings plans are applied to covered usage, the costs for that savings plan appear as usage charges. The unused portion appears as a fee.
```

File: content/en/cloud_cost_management/aws.md [hunk 5]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+
+Not all one-time fees are amortized across the billing period. For example, the unused portion of a Reserved Instance Fee typically appears on the first of the month.
+
+**Unblended** cost metrics show all charges on the date that they were incurred. This is also called _cash basis_. These metrics match the AWS invoice exactly.
```
Comment (urseberry) [inline content/en/cloud_cost_management/aws.md:152]:
```suggestion
**Unblended** cost metrics show all charges on the date that they are incurred. Another name for unblended cost metrics is _cash basis_. These metrics match the AWS invoice exactly.
```

**Review by urseberry — COMMENTED** [review]
I am part way through reviewing - will finish later today or tomorrow.

File: content/en/cloud_cost_management/aws.md (hunk 4 above)
Comment (enemyghost) [reply content/en/cloud_cost_management/aws.md:148]: Not sure about this one, since we actually don't call this anywhere in our own product... it's more like "this type of cost reporting is also known as accrual basis / cash basis".

File: content/en/cloud_cost_management/aws.md [hunk 6]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+
+### Net costs
+**Net** costs apply private discounts directly to usage. The cost of usage for a specific resource represents the effective cost after all savings are realized.
+
+Other metrics (without the `net` keyword) show private discounts as separate, negative-valued line items with no attribution tags. Rather than attributing the discounts to actual usage, they are treated as offsets to total cost.
```
Comment (guilhermevrs) [inline content/en/cloud_cost_management/aws.md:157]: **💭 thought:** ‏Perhaps we should refrain of talking about keywords and suffixes. I believe our overall goal is to hide the name of the metrics (once we migrate the new searchbar to dashboards).
If we don't have a better solution now, we can revisit that later

Comment (enemyghost) [reply content/en/cloud_cost_management/aws.md:157]: great point, I will try to revise this. I will leave the full metric names in the Metric Summary table for now, and we can modify that when they're no longer visible if we want.

Though, I'm guessing you will still see them in string-query edit mode?

File: content/en/cloud_cost_management/aws.md (hunk 4 above)
Comment (urseberry) [reply content/en/cloud_cost_management/aws.md]: Got it. Thanks for explaining.

File: content/en/cloud_cost_management/aws.md [hunk 7]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+|---|---|---|
+|lineItem/resourceId|i-12345678a9b12cd3e|line_item/resource_id:i-12345678a9b12cd3e|
+|product/region|us-east-1|product/region:us-east-1|
+|product/usagetype|DataTransfer-Regional-Bytes|product/usagetype:DataTransfer-Regional-Bytes|
+
+### AWS resource tags
+
+[AWS resource tags][12] are user-defined tags that appear in the AWS console when viewing a particular resource, like an EC2 instance or S3 bucket.
+
+When the Datadog AWS integration is enabled, Datadog automatically collects resource tags for most AWS resources. These tags are applied to all costs found in the CUR for a given resource. Resource tags are retrieved regularly and are applied to cost data starting from the day they are created or modified. Historical tag values are not overwritten when tags change.
```
Comment (urseberry) [inline content/en/cloud_cost_management/aws.md:228]:
```suggestion
When you enable the Datadog AWS integration, Datadog automatically collects resource tags for most AWS resources. These tags are applied to all costs found in the CUR for a given resource. Resource tags are retrieved regularly and are applied to cost data starting from the day they are created or modified. Historical tag values are not overwritten when tags change.
```

File: content/en/cloud_cost_management/aws.md [hunk 8]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+**Net amortized** costs provide the most accurate representation for cost allocation, with all savings applied directly to usage. Net cost metrics are available if your AWS account has privately negotiated enterprise discounts. If your account doesn't have net metrics available, use **amortized** cost instead.
+
+### Container allocation
+**Container allocation** metrics contain all of the same costs as the AWS metrics, but with additional breakdowns and insights for container workloads. See [container cost allocation][11] for more details.
+
+### Example
+Here is an example scenario to demonstrate how these cost types behave. Imagine you have:
+- an EC2 instance running for one hour with the cost of $3/compute-hour.
+- a savings plan which covers this instance type and hour purchased for $2.
+- a negotiated discount of 10% on top of all other discounts.
```
[the remark this answers is not in the document]
Comment (enemyghost) [reply content/en/cloud_cost_management/aws.md:168]: Your interpretation is correct, though i was trying to skip over the complexity of "purchased at a cost of $X". Updated to remove - does it make sense?

File: content/en/cloud_cost_management/aws.md [hunk 9]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+
+[AWS resource tags][12] are user-defined tags that appear in the AWS console when viewing a particular resource, like an EC2 instance or S3 bucket.
+
+When the Datadog AWS integration is enabled, Datadog automatically collects resource tags for most AWS resources. These tags are applied to all costs found in the CUR for a given resource. Resource tags are retrieved regularly and are applied to cost data starting from the day they are created or modified. Historical tag values are not overwritten when tags change.
+
+If the AWS integration is not enabled, you can enable resource tag enrichment by activating [cost allocation tags][13] in AWS billing. This allows you to select a subset of resource tag keys to include as columns in the AWS CUR. Datadog automatically includes those columns as tags when processing the CUR.
+
+### AWS organization and account tags
+AWS Organizations supports [user-defined tags][14] on organizational units and accounts. Datadog automatically fetches and applies these tags to cost data. Account tags are applied to all usage associated with those accounts. Organization tags are applied to all billing data for the matching payer account.
```
Comment (urseberry) [inline content/en/cloud_cost_management/aws.md:233]:
```suggestion
AWS Organizations support [user-defined tags][14] on organizational units and accounts. Datadog automatically fetches and applies these tags to cost data. Account tags are applied to all usage associated with those accounts. Organization tags are applied to all billing data for the matching payer account.
```

File: content/en/cloud_cost_management/aws.md [hunk 10]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+**Amortized** cost metrics distribute some pre-payments (such as up-front savings plan fees) across the entire discount term. This is also called _accrual basis_. As savings plans are applied to covered usage, the costs for that savings plan appear as usage charges. The unused portion appears as a fee.
+
+Not all one-time fees are amortized across the billing period. For example, the unused portion of a Reserved Instance Fee typically appears on the first of the month.
+
+**Unblended** cost metrics show all charges on the date that they are incurred. This is also called _cash basis_. These metrics match the AWS invoice exactly.
+
+### Net costs
+**Net** costs apply private discounts directly to usage. The cost of usage for a specific resource represents the effective cost after all savings are realized.
+
+Non-net metrics show private discounts as separate, negative-valued line items with no attribution tags. Rather than attributing the discounts to actual usage, they are treated as offsets to total cost.
```
[the remark this answers is not in the document]
Comment (enemyghost) [reply content/en/cloud_cost_management/aws.md:157]: @urseberry  Tried to reword this a bit. Had some feedback that "non-net" is not really a term we want to introduce since it's a bit confusing. Wdyt?

File: content/en/cloud_cost_management/aws.md [hunk 11]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+If the AWS integration is not enabled, you can enable resource tag enrichment by activating [cost allocation tags][13] in AWS billing. This allows you to select a subset of resource tag keys to include as columns in the AWS CUR. Datadog automatically includes those columns as tags when processing the CUR.
+
+### AWS organization and account tags
+AWS Organizations supports [user-defined tags][14] on organizational units and accounts. Datadog automatically fetches and applies these tags to cost data. Account tags are applied to all usage associated with those accounts. Organization tags are applied to all billing data for the matching payer account.
+
+_Requires the Datadog AWS Integration on the organization account._
+
+### AWS integration tags
+
+These are tags set on the AWS Integration tile in the Datadog integrations page. They are applied to all costs found in the CUR for the associated AWS account.
```
Comment (urseberry) [inline content/en/cloud_cost_management/aws.md:239]:
```suggestion
AWS integration tags are tags set on the AWS Integration tile in the Datadog integrations page. They are applied to all costs found in the CUR for the associated AWS account.
```

File: content/en/cloud_cost_management/aws.md [hunk 12]
```diff
@@ -202,8 +294,15 @@ The following out-of-the-box tags are available:
 | elb (application, gateway, network) | `loadbalancer`|
 | elb (all other costs) | `loadbalancername` |
 
-For tags on containerized environments, see [Container Cost Allocation][11].
+### Container orchestrators
+
+Container cost allocation adds tags from the workloads incurring cost. This includes tags from Kubernetes pods and nodes and ECS tasks and containers.
```
Comment (urseberry) [inline content/en/cloud_cost_management/aws.md:299]:
```suggestion
Container cost allocation adds tags from the workloads incurring cost. Examples include tags from Kubernetes pods and nodes and ECS tasks and containers.
```

File: content/en/cloud_cost_management/aws.md [hunk 13]
```diff
@@ -202,8 +294,15 @@ The following out-of-the-box tags are available:
...
+
+_Requires [container cost allocation][11], and applies only to shared.resources.allocated metrics._
```
Comment (urseberry) [inline content/en/cloud_cost_management/aws.md:301]:
```suggestion
_Requires [container cost allocation][11], and applies only to `shared.resources.allocated` metrics._
```

**Review by urseberry — COMMENTED** [review]
Whew, all done with my text review. Nice job on this - I appreciate your precise language. Describing the details of these different cost calculations is a difficult task.

I'll go back and do one more pass to check the links, which I just noticed, but I assume those are probably going to be fine.

File: content/en/cloud_cost_management/aws.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (urseberry) [reply content/en/cloud_cost_management/aws.md:168]: Yes, it makes sense, thanks!

File: content/en/cloud_cost_management/aws.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (urseberry) [reply content/en/cloud_cost_management/aws.md:157]: Yes, your latest wording looks great.

**Comment by enemyghost** [comment]
Thanks @urseberry for the thorough review, really appreciate your edits. I think at least one more subject matter expert will review the content, then I'll get your final sign off.

File: content/en/cloud_cost_management/aws.md [hunk 14]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+- Container workload tags
+- Tag pipelines
 
-### Tag enrichment
+### Cost and Usage Report columns
 
-Datadog adds out-of-the-box tags to the ingested cost data to help you further break down and allocate your costs. These tags are derived from your [Cost and Usage Report (CUR)][6].
+All string-valued columns from the AWS [Cost and Usage Report (CUR)][6] are added as tags on cost metrics.
+
+To ensure consistency, Datadog normalizes tag keys using underscores and lower case. For example, the CUR column `lineItem/resourceId` maps to the tag key `line_item/resource_id`. Tag values are generally unmodified - maintaining exact casing and most special characters.
```
Comment (cnharris10) [inline content/en/cloud_cost_management/aws.md:214]:
```suggestion
To ensure consistency, Datadog normalizes tag keys using underscores and lower case. For example, the CUR column `lineItem/ResourceId` maps to the tag key `line_item/resource_id`. Tag values are generally unmodified - maintaining exact casing and most special characters.
```

File: content/en/cloud_cost_management/aws.md [hunk 15]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+
+**Examples:**
+
+|CUR Column|CUR Value|Cloud Cost Tag|
+|---|---|---|
+|lineItem/resourceId|i-12345678a9b12cd3e|line_item/resource_id:i-12345678a9b12cd3e|
```
Comment (cnharris10) [inline content/en/cloud_cost_management/aws.md:220]:
```suggestion
|lineItem/ResourceId|i-12345678a9b12cd3e|line_item/resource_id:i-12345678a9b12cd3e|
```

**Review by cnharris10 — CHANGES_REQUESTED** [review]
Let me know your thoughts on these suggestions.

File: content/en/cloud_cost_management/aws.md [hunk 16]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+**Amortized** cost metrics distribute some pre-payments (such as up-front savings plan fees) across the entire discount term. This is also called _accrual basis_. As savings plans are applied to covered usage, the costs for that savings plan appear as usage charges. The unused portion appears as a fee.
+
+Not all one-time fees are amortized across the billing period. For example, the unused portion of a Reserved Instance Fee typically appears on the first of the month.
+
+**Unblended** cost metrics show all charges on the date that they are incurred. This is also called _cash basis_. These metrics match the AWS invoice exactly.
+
+### Net costs
+**Net** costs apply private discounts directly to usage. The cost of usage for a specific resource represents the effective cost after all savings are realized.
+
+In contrast, other metrics show private discounts as separate, negative-valued line items with no attribution tags. Rather than attributing the discounts directly to usage, those metrics subtract discounts from the total cost.
```
[the remark this answers is not in the document]
Comment (enemyghost) [reply content/en/cloud_cost_management/aws.md:157]: i think `some` is still a good edit, though I did initially mean resource id attribution.

File: content/en/cloud_cost_management/aws.md [hunk 17]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+Visualize your ingested data using out-of-the-box cost types. Cloud cost metric names contain various terms that describe how they are calculated.
 
-| Cost Type            | Description           |
+### On-demand
+**On-demand** costs are the costs of usage at the public, on-demand rate published by AWS. This excludes all savings plans, reservations, discounts, taxes, and fees.
+
+On-demand costs do not accurately represent what you pay AWS. Do not use this metric to estimate your actual costs.
+
+### Amortized and unblended costs
+**Amortized** cost metrics distribute some pre-payments (such as up-front savings plan fees) across the entire discount term. This is also called _accrual basis_. As savings plans are applied to covered usage, the costs for that savings plan appear as usage charges. The unused portion appears as a fee.
```
[the remark this answers is not in the document]
Comment (enemyghost) [reply content/en/cloud_cost_management/aws.md:148]: agree on both counts 
- i will switch to "cost basis"
- reworded your edit a bit, lmk if it makes sense.

[the remark this answers is not in the document]
Comment (enemyghost) [reply content/en/cloud_cost_management/aws.md:148]: with this edit, maybe i will delete the next paragraph, which I think is now confusing after this description.

File: content/en/cloud_cost_management/aws.md [hunk 18]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+## Cost types
 
-You can visualize your ingested data using the following cost types:
+Visualize your ingested data using out-of-the-box cost types. Cloud cost metric names contain various terms that describe how they are calculated.
 
-| Cost Type            | Description           |
+### On-demand
+**On-demand** costs are the costs of usage at the public, on-demand rate published by AWS. This excludes all savings plans, reservations, discounts, taxes, and fees.
+
+On-demand costs do not accurately represent what you pay AWS. Do not use this metric to estimate your actual costs.
```
[the remark this answers is not in the document]
Comment (enemyghost) [reply content/en/cloud_cost_management/aws.md:145]: I didn't want to introduce the term "unblended" here yet, and I think this edited sentence is kind of confusing. In this section, I try to provide light guidance, not a complete description with all caveats. Are you concerned about saying "they do not reflect actual costs" when in some cases they do?

What about something like:

In most cases, on-demand costs are not a reliable source to estimate actual costs.

File: content/en/cloud_cost_management/aws.md [hunk 19]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+
+### Amortized and unblended costs
+**Amortized** cost metrics distribute some pre-payments (such as up-front savings plan fees) across the entire discount term. This is also called _accrual basis_. As savings plans are applied to covered usage, the costs for that savings plan appear as usage charges. The unused portion appears as a fee.
+
+Not all one-time fees are amortized across the billing period. For example, the unused portion of a Reserved Instance Fee is typically backdated to the first of the month.
+
```
Comment (enemyghost) [inline content/en/cloud_cost_management/aws.md:151]:
```suggestion
```

File: content/en/cloud_cost_management/aws.md [hunk 20]
```diff
@@ -135,20 +135,111 @@ Attach the new S3 policy to the Datadog integration role.
...
+**Unblended** cost metrics show all charges on the date that they are incurred. This is also called _cash basis_. These metrics match the AWS invoice exactly once a month is finalized.
```
Comment (enemyghost) [inline content/en/cloud_cost_management/aws.md:152]:
```suggestion
**Unblended** cost metrics show all charges on the date that they are incurred. This is also called _cost basis_. These metrics match the AWS invoice exactly after a month is finalized.
```

**Review by urseberry — APPROVED** [review]
Looks like this is good to merge after @cnharris10 approves?

**Review by cnharris10 — APPROVED** [review]
I still find the Cost metric summary table hard to follow, particularly since the `Usage` and `Discount` columns show effective costs, but the `Savings Plan` column does not.  I won't block on this as long as others feel differently.
